### ISSUE TITLE
Quasar Tree returning too much data when all items are selected

### ISSUE THREAD

**Issue by HouseSitSearch** [quasarframework/quasar#8008]
I'm using the Quasar Tree to have a compact control in which to include Regions, Countries, and States from around the world, for a search criteria. I'm currently using the vue-treeselect to do this
https://vue-treeselect.js.org/#node

However, I would prefer to use the Quasar Tree, but the data that is being returned is far more verbose than vue-treeselect, and results in a significantly more complicated SQL query when returned to the server.

vue-treeselect, simply returns ["All"] when the root node is checked, or ["Europe", "NorthAmerica"] when only those items are selected.

The problem I'm having with Quasar's Tree component, is that it's returning every Country and State in Europe and North America instead. Obviously this includes 50 states for the USA alone, so a lot more data is being generated than necessary. And instead of the server simply saying "where Region = 'NorthAmerica'" it is instead having to do "where State='Alabama' or State='Alaska' or State='Arizona'" etc., which is very inefficient, and puts unnecessary stress on the database server.

The vue-treeselect component achieves this using a "valueConsistsOf" prop.
https://vue-treeselect.js.org/#prevent-value-combining

If you quickly play with this link, you can see how the Quasar Tree effectively uses "Leaf priority", whereas it would be much more efficient for most people to use the "Branch priority" option, which is the default option for vue-treeselect.

**Comment by HouseSitSearch** [comment]
As an example of the data produced, here's the data from the 2 components when only the "Europe" and "North America" nodes are checked in the tree. NOTE: This becomes much worse for the Quasar Tree when the root node is checked, as every leaf node is then returned.

vue-treeselect
["RG=EUR","RG=NAM"]

Quasar Tree
["RG=EUR&CN=AUT","RG=EUR&CN=BEL","RG=EUR&CN=BGR","RG=EUR&CN=HRV","RG=EUR&CN=CYP","RG=EUR&CN=DNK","RG=EUR&CN=FRA","RG=EUR&CN=DEU","RG=EUR&CN=GRC","RG=EUR&CN=IRL","RG=EUR&CN=ITA","RG=EUR&CN=MLT","RG=EUR&CN=MNE","RG=EUR&CN=NLD","RG=EUR&CN=NOR","RG=EUR&CN=PRT","RG=EUR&CN=ROU","RG=EUR&CN=SVK","RG=EUR&CN=ESP","RG=EUR&CN=SWE","RG=EUR&CN=CHE","RG=EUR&CN=TUR","RG=EUR&CN=GBR&ST=ENG","RG=EUR&CN=GBR&ST=SCO","RG=EUR&CN=GBR&ST=WAL","RG=NAM&CN=CAN&ST=","RG=NAM&CN=CAN&ST=AB","RG=NAM&CN=CAN&ST=BC","RG=NAM&CN=CAN&ST=ON","RG=NAM&CN=CAN&ST=QC","RG=NAM&CN=MEX","RG=NAM&CN=USA&ST=","RG=NAM&CN=USA&ST=AZ","RG=NAM&CN=USA&ST=AR","RG=NAM&CN=USA&ST=CA","RG=NAM&CN=USA&ST=CO","RG=NAM&CN=USA&ST=CT","RG=NAM&CN=USA&ST=FL","RG=NAM&CN=USA&ST=GA","RG=NAM&CN=USA&ST=HI","RG=NAM&CN=USA&ST=ID","RG=NAM&CN=USA&ST=IL","RG=NAM&CN=USA&ST=IN","RG=NAM&CN=USA&ST=IA","RG=NAM&CN=USA&ST=KS","RG=NAM&CN=USA&ST=MD","RG=NAM&CN=USA&ST=MA","RG=NAM&CN=USA&ST=MI","RG=NAM&CN=USA&ST=MN","RG=NAM&CN=USA&ST=MT","RG=NAM&CN=USA&ST=NE","RG=NAM&CN=USA&ST=NV","RG=NAM&CN=USA&ST=NJ","RG=NAM&CN=USA&ST=NM","RG=NAM&CN=USA&ST=NY","RG=NAM&CN=USA&ST=NC","RG=NAM&CN=USA&ST=OH","RG=NAM&CN=USA&ST=OK","RG=NAM&CN=USA&ST=OR","RG=NAM&CN=USA&ST=PA","RG=NAM&CN=USA&ST=SC","RG=NAM&CN=USA&ST=SD","RG=NAM&CN=USA&ST=TN","RG=NAM&CN=USA&ST=TX","RG=NAM&CN=USA&ST=UT","RG=NAM&CN=USA&ST=VA","RG=NAM&CN=USA&ST=WA","RG=NAM&CN=USA&ST=WI"]

**Comment by hawkeye64** [comment]
@HouseSitSearch Thanks for the suggestion :heart: 
I think this is a good idea, but would introduce a breaking change for existing users.
I think we should take it under consideration for the next major release.

**Comment by rstoenescu** [comment]
A `combine` prop (with values for each possible strategy) can be added in Quasar v1. Don't need to wait for the next major release.

**Comment by hawkeye64** [comment]
For instance?
combine="all" (default)
combine="flatten"
Any other strategies ???

**Comment by HouseSitSearch** [comment]
I agree with hawkeye64. If you add a new property and make the default for that new property have the same functionality as the existing code, then there is no breaking change.

**Comment by HouseSitSearch** [comment]
I've never done any development work for Quasar, but is it possible I could contribute to help implement this?

I don't know where to start though, so I'd need some help getting setup.

**Comment by hawkeye64** [comment]
@HouseSitSearch DM me on Discord and I'll help you get set up

**Comment by philliskiragu** [comment]
@HouseSitSearch Did you manage to include the `All` option to the tree?

**Comment by hawkeye64** [comment]
@philliskiragu As far as I know, it has not been done. Maybe @rstoenescu will take it under advisement for upcoming v2 release?

**Comment by petr-panek** [comment]
I'd like to add my wish for this functionality. 

vue-treeselect doesn't work properly on Vue 3 (generate some warnings) and it's also not maintained anymore. Due this we have currently big issue to upgrade to Quasar 2/Vue 3.
